SurveyApp/Modules/Login/LoginViewController.swift

```diff
@@ -49,6 +56,24 @@ extension LoginViewController: LoginViewInput {
         setUpLayout()
         setUpViews()
     }
+    
+    func beginAnimation() {
+        UIView.animate(withDuration: 1.0, delay: 0.0, options: .transitionCurlUp) { [weak self] in
```

Comment:
here's a good read https://medium.com/flawless-app-stories/you-dont-always-need-weak-self-a778bec505ef. There's no need to reference weak self here as `UIView.animate` will eventually release self.